docs/src/pages/components/data-grid/demo/FullFeauredDemo.tsx

```diff
@@ -0,0 +1,270 @@
+  const [typeState, setType] = React.useState(type);
+  const [selectedPaginationValue, setSelectedPaginationValue] = React.useState(-1);
+  const [activeTheme, setActiveTheme] = React.useState(theme);
+
+  const handleSizeChange = React.useCallback((event) => {
+    setSize(Number(event.target.value));
+  }, []);
+
+  const handleDatasetChange = React.useCallback((event) => {
+    setType(event.target.value);
+  }, []);
+
+  const handlePaginationChange = React.useCallback((event) => {
+    setSelectedPaginationValue(event.target.value);
+  }, []);
+
+  const handleThemeChange = React.useCallback((event) => {
+    setActiveTheme(event.target.value);
+  }, []);
+
+  const handleApplyChanges = React.useCallback(() => {
+    onApply({
+      size: sizeState,
+      type: typeState,
+      pagesize: selectedPaginationValue,
+      theme: activeTheme,
+    });
+  }, [sizeState, typeState, selectedPaginationValue, activeTheme, onApply]);
+
+  return (
+    <FormGroup className="MuiFormGroup-options" row>
+      <FormControl size="small" component="fieldset">
+        <FormLabel component="legend">Dataset</FormLabel>
+        <Select value={typeState} onChange={handleDatasetChange}>
+          <MenuItem value="Employee">Employee</MenuItem>
+          <MenuItem value="Commodity">Commodity</MenuItem>
+        </Select>
+      </FormControl>
+      <FormControl size="small" component="fieldset">
+        <FormLabel component="legend">Rows</FormLabel>
+        <Select value={sizeState} onChange={handleSizeChange}>
+          <MenuItem value={100}>100</MenuItem>
+          <MenuItem value={1000}>{Number(1000).toLocaleString()}</MenuItem>
+          <MenuItem value={10000}>{Number(10000).toLocaleString()}</MenuItem>
+          <MenuItem value={100000}>{Number(100000).toLocaleString()}</MenuItem>
+        </Select>
+      </FormControl>
+      <FormControl size="small" component="fieldset">
+        <FormLabel component="legend">Page Size</FormLabel>
+        <Select value={selectedPaginationValue} onChange={handlePaginationChange}>
+          <MenuItem value={-1}>off</MenuItem>
+          <MenuItem value={0}>auto</MenuItem>
+          <MenuItem value={25}>25</MenuItem>
+          <MenuItem value={100}>100</MenuItem>
+          <MenuItem value={1000}>{Number(1000).toLocaleString()}</MenuItem>
+        </Select>
+      </FormControl>
+      <FormControl size="small" component="fieldset">
+        <FormLabel component="legend">Theme</FormLabel>
+        <Select value={activeTheme} onChange={handleThemeChange}>
+          <MenuItem value="default">Default Theme</MenuItem>
+          <MenuItem value="antd">Antd Design</MenuItem>
+        </Select>
+      </FormControl>
+      <Button
+        size="small"
+        variant="outlined"
+        color="primary"
+        onClick={handleApplyChanges}
+      >
+        <KeyboardArrowRightIcon fontSize="small" /> Apply
+      </Button>
+    </FormGroup>
+  );
+}
+
+export default function FullFeaturedDemo() {
```

Comment:
Can't this file be split like any applications?
It will be difficult to scale if we have to put everything in one file :D

Reply:
It's possible. If we do so, we should disable the export to codesandbox since imports aren't followed.

Reply:
I had the same thought about this, but then I saw other big demos in the core and I think it is valuable for people to be able to open it directly in **codesandbox**. No preference here - we can keep it all together for now and split them if it becomes too big. Currently, the file is at about 270 lines which is a lot but still acceptable considering that 100 lines are styles.

Reply:
Regarding the option to add multifile support to the codesandbox export, I doubt the ROI is positive.

Reply:
It's  a big demo, not sure this one needs to be open in codesandbox

Reply:
But then how would they copy/paste it or play with the code easily?

Reply:
To add on Damien points, we often display access to the source in the main documentation for playgrounds. It's usually not really useful.

Maybe we can start with a single file, after a couple of days, look at the Google Analytics events to see if developers care about accessing the source?